src/time-map.scm

```diff
@@ -1,8 +1,82 @@
-(use-modules (opencog) (opencog atom-types) (opencog ato pointmem)); needed for mapsi
+(use-modules (opencog) (opencog atom-types) (opencog eva-behavior))
+(use-modules (opencog ato pointmem)); needed for mapsi
 (use-modules (opencog python))
 ;;initialize octomap with 15hz, 10 second or 150 frames buffer ; 1 cm spatial resolution
 (create-map "faces" 0.01 66 150) (step-time-unit "faces")(auto-step-time-on "faces")
+(create-map "sounds" 0.01 100 100) (step-time-unit "sounds")(auto-step-time-on "sounds")
+
+;;returns null string if atom not found, number x y z string if okay
+;;these functions assume only one location for one atom in a map at a time
+(define (get-past-xyz map-name id-node elapse)
+		(let* ((loc-atom (get-past-locs-ato map-name id-node elapse) ))
+			(if (equal? (cog-atom (cog-undefined-handle)) loc-atom) ""
+				(let* ((loc-link (car (cog-outgoing-set loc-atom)))
+					(xx (number->string (loc-link-x loc-link)))
+					(yy (number->string (loc-link-y loc-link)))
+					(zz (number->string (loc-link-z loc-link))))
+						(string-append xx " " yy " " zz))
+			)
+		)
+)
+;;returns null string if atom not found, number x y z string if okay
+(define (get-xyz map-name id-node)
+		(let* ((loc-atom (get-locs-ato map-name id-node) ))
+			(if (equal? (cog-atom (cog-undefined-handle)) loc-atom) ""
+				(let* ((loc-link (car (cog-outgoing-set loc-atom)))
+					(xx (number->string (loc-link-x loc-link)))
+					(yy (number->string (loc-link-y loc-link)))
+					(zz (number->string (loc-link-z loc-link))))
+						(string-append xx " " yy " " zz))
```

Comment:
there's an awful lot of string to number, number to string conversions in this code.  Wouldn't it be easier to to avoid all these conversions?